h2o-genmodel-extensions/jgrapht/src/main/java/hex/genmodel/tools/JgraphtPrintMojo.java

```diff
@@ -0,0 +1,227 @@
+package hex.genmodel.tools;
+
+import com.mxgraph.layout.mxCompactTreeLayout;
+import com.mxgraph.layout.mxEdgeLabelLayout;
+import com.mxgraph.layout.mxIGraphLayout;
+import com.mxgraph.util.mxCellRenderer;
+import hex.genmodel.algos.tree.ConvertTreeOptions;
+import hex.genmodel.algos.tree.SharedTreeGraph;
+import hex.genmodel.algos.tree.SharedTreeSubgraph;
+import hex.genmodel.algos.tree.TreeBackedMojoModel;
+import org.jgrapht.ext.JGraphXAdapter;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.DirectedMultigraph;
+import org.jgrapht.io.*;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.Map;
+
+public class JgraphtPrintMojo extends PrintMojo implements MojoPrinter {
+
+    public static void main(String[] args) {
+        MojoPrinter mojoPrinter;
+        if ("png".equals(getFormat(args))){
+            mojoPrinter = new JgraphtPrintMojo();
+        } else {
+            mojoPrinter = new PrintMojo();
+        }
+        mojoPrinter.parseArgs(args);
+
+        // Run the main program
+        try {
+            mojoPrinter.run();
+        } catch (Exception e) {
+            e.printStackTrace();
+            System.exit(2);
+        }
+        
+        // Success
+        System.exit(0);
+    }
+    
+    @Override
+    protected void validateArgs() {
+        if (genModel == null) {
```

Comment:
this method is a copy of the overriden method, it should not be, the validation that output!=png for jdk7 can be done in differently

Reply:
fixed